runner/full.go

```diff
@@ -17,6 +19,55 @@ import (
 	"github.com/pganalyze/collector/util"
 )
 
+const MinSupportedLogMinDurationStatement = 100
+
+func enforceLogCollectionLimits(server state.Server, transientState state.TransientState) {
```

Comment:
I wonder if in addition to calling this from the full snapshots, we should also call it once when the collector reloads/restarts (as you had this initially). We'd have to pull in the settings with an extra query for that, but I think thats okay on startup.

That would ensure that you can always clear this behavior by doing that action, vs having to wait up to 10 minutes to get a full snapshot in.

Reply:
@lfittl reading through `main.go`, the [server array is rebuilt](https://github.com/pganalyze/collector/blob/master/main.go#L75) every reload, no? So this new approach, of storing that log-disabled state on the server object itself, rather than the persisted objects solves this problem, right? (Assuming we're okay with <=10m of high log volume after every restart; we could still try to pull up the check to try to avoid that.)

Reply:
We do, however, need to avoid logging so frequently on this:

```
2020/10/19 16:09:39 I [server1] Testing log collection (local)...
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:39 W [server1] Skipping logs: log_statement is set to unsupported value 'all'
2020/10/19 16:09:49 E [server1] ERROR - Could not tail logs for server: Timeout
```
(And fix that tail issue.) Looking into this.

Reply:
Ok, with properly handling this in `TestLogsForAllServers`, we avoid the warnings in the test code path and this becomes:

```
2020/10/19 16:39:21 I [server1] Testing statistics collection...
2020/10/19 16:39:24 I [server1]   Test submission successful (645 KB received, server 6ayj3psrubag7gwuzv7sbwnq7a)
2020/10/19 16:39:24 I [server1] Testing activity snapshots...
2020/10/19 16:39:24 I [server1]   Test submission successful (5.07 KB received, server 6ayj3psrubag7gwuzv7sbwnq7a)
2020/10/19 16:39:24 W [server1] WARNING - Configuration issue: log_statement is set to unsupported value 'all'
2020/10/19 16:39:24 W [server1]   Log collection will be disabled for this server
```
Still looking at avoiding that warning in an actual run.

Reply:
Log tails seem to be a third place where we need to check this, right? That doesn't seem to be handled by either the log stream subscription handling or the file downloads. Also, I asked this before but didn't follow up: should we be clearing the AWS log marker when we enable collection after it has been disabled? I think without that we'd process all the log files that were generated while the naughty settings were in place, right?

Reply:
Ok, so in 0259620, I tried to bite the bullet and change all the pass-by-value to pass-by reference. The log-disabling now seems to work. I think according to [this SO discussion](https://stackoverflow.com/questions/23542989/pointers-vs-values-in-parameters-and-return-values) this is a reasonable change (pretty sure the `Server` struct is too big for there to be any performance benefit to pass-by-value, and while avoiding modification by callees is nice to have, I'm not that worried about that). Thoughts?

Reply:
Oh come to think of it, we can just warn when we determine whether logs are disabled based on settings--that will just happens once per full snapshot which seems reasonable. I'll do that.

Reply:
Ended up doing this; we could easily warn just once per collector reload as well, but warning once per full snapshot doesn't seem too verbose?

Reply:
> Ended up doing this; we could easily warn just once per collector reload as well, but warning once per full snapshot doesn't seem too verbose?

Yeah, that seems fine - if we get complaints because this is too verbose we could adjust, but I think err-ing on the side of warning frequently is a good thing (to make it clear why that part of the product isn't working).